template-bot-master-csharp/utility/WikipediaComposeExtension.cs

```
@@ -326,69 +326,53 @@ public static ComposeExtensionResponse GetSignin(ComposeExtensionResponse compos
+        public async Task<WikiResult> SearchWiki(string queryParameter, ComposeExtensionQuery composeExtensionQuery)
         {
             string searchApiUrl = SearchApiUrlFormat.Replace("[keyword]", queryParameter);
             searchApiUrl = searchApiUrl.Replace("[limit]", composeExtensionQuery.QueryOptions.Count + "");
             searchApiUrl = searchApiUrl.Replace("[offset]", composeExtensionQuery.QueryOptions.Skip + "");
-            WikiResult wikiResult = null;
-
-            // call Wikipedia API to search
-            try
-            {
-                HttpWebRequest myRequest = (HttpWebRequest)WebRequest.Create(searchApiUrl);
-                using (HttpWebResponse response = (HttpWebResponse)myRequest.GetResponse())
-                {
-                    string ResponseText;
-                    using (StreamReader reader = new StreamReader(response.GetResponseStream()))
-                    {
-                        ResponseText = reader.ReadToEnd();
-                        wikiResult = JsonConvert.DeserializeObject<WikiResult>(ResponseText);
-                    }
-                }
-            }
-            catch (Exception ex)
-            {
 
-            }
-            return wikiResult;
+            Uri apiUrl = new UriBuilder(searchApiUrl).Uri;
+            return await ProcessRequest<WikiResult>(apiUrl);
         }
 
-        public static ImageResult SearchWikiImage(Search wikiSearch)
+        public async Task<ImageResult> SearchWikiImage(Search wikiSearch)
         {
-            ImageResult imageResult = null;
             // a separate API call to Wikipedia is needed to fetch the page image, if it exists
             string imageApiUrl = ImageApiUrlFormat.Replace("[title]", wikiSearch.title);
 
+            Uri apiUrl = new UriBuilder(imageApiUrl).Uri;
+            return await ProcessRequest<ImageResult>(apiUrl);
+        }
+
+        private async Task<T> ProcessRequest<T>(Uri uri)
+        {
+            string json;
```

Can't this be in the try/catch?
Also, don't declare unnecessary variables.'
Finally, converting JSON deserialization failure to ArgumentException doesn't make sense. There's nothing wrong with the argument. Just let the JsonException propagate.

var json = await client.GetStringAsync(uri);
return JsonConvert.DeserializeObject<T>(json);

---

Done

---

Done.